registry/txt_test.go

```diff
@@ -114,7 +114,7 @@ func testTXTRegistryRecordsPrefixed(t *testing.T) {
 			newEndpointWithOwner("dualstack.test-zone.example.org", "1.1.1.1", endpoint.RecordTypeA, ""),
 			newEndpointWithOwner("txt.dualstack.test-zone.example.org", "\"heritage=external-dns,external-dns/owner=owner\"", endpoint.RecordTypeTXT, ""),
 			newEndpointWithOwner("dualstack.test-zone.example.org", "2001:DB8::1", endpoint.RecordTypeAAAA, ""),
-			newEndpointWithOwner("aaaa-txt.dualstack.test-zone.example.org", "\"heritage=external-dns,external-dns/owner=owner-2\"", endpoint.RecordTypeTXT, ""),
+			newEndpointWithOwner("txt.aaaa-dualstack.test-zone.example.org", "\"heritage=external-dns,external-dns/owner=owner-2\"", endpoint.RecordTypeTXT, ""),
```

Comment:
Isn't this an incompatible change? People with existing AAAA ownership records have them created with `aaaa-txt.`, not with what you're changing it to.

Reply:
my change is not modifying how this are generated, that test it simply wrong

Reply:
> Additionally, this PR fixes the testTXTRegistryRecordsPrefixed test, which previously operated under the incorrect assumption that the output of toNewTXTName with the prefix txt. for dualstack.test-zone.example.org would be aaaa-txt.dualstack.test-zone.example.org. In reality when not using a templated record type, the type will be prepended to the domain rather than the affix.
> ```go
> DNSName := strings.SplitN(endpointDNSName, ".", 2)
> 
> if !pr.recordTypeInAffix() {
> 	DNSName[0] = recordT + DNSName[0]
> }
> ```

Reply:
I withdraw this comment.